.github/workflows/ci.yml

```diff
@@ -1,103 +1,163 @@
-name: Main
+name: CI
+
 on:
   push:
-    branches:
-      - main
+    branches: [main]
   pull_request:
+    branches: [main]
+
+env:
+  BUNDLE_APP_CONFIG: .bundle
 
 jobs:
   ruby-lint:
+    name: Ruby Lint (Ruby ${{ matrix.ruby }})
     runs-on: ubuntu-latest
-    env:
-      BUNDLE_APP_CONFIG: .bundle
+    strategy:
+      matrix:
+        ruby: ['3.4']
     steps:
-      - uses: actions/checkout@v4
-      - uses: ruby/setup-ruby@v1
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
         with:
-          ruby-version: '3.2'
-          bundler-cache: true
-      - run: bundle exec rubocop
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: false
+
+      - name: Configure bundler
+        run: |
+          bundle config set --local path 'vendor/bundle'
+          bundle config set --local frozen false
+
+      - name: Install dependencies
+        run: bundle install
+
+      - name: Run RuboCop
+        run: bundle exec rubocop
+
   ruby-test:
+    name: Ruby Test (Ruby ${{ matrix.ruby }}, GraphQL ${{ matrix.graphql }})
     runs-on: ubuntu-latest
-    env:
-      BUNDLE_APP_CONFIG: .bundle
     strategy:
       fail-fast: false
       matrix:
-        appraisal: [ 'graphql-1.10', 'graphql-1.11', 'graphql-1.12', 'graphql-1.13', 'graphql-2.0' ]
-        ruby: [ '3.0', '3.1', '3.2' ]
-        exclude:
-          - appraisal: 'graphql-1.10'
-            ruby: '3.2'
-          - appraisal: 'graphql-1.11'
-            ruby: '3.2'
-          - appraisal: 'graphql-1.12'
-            ruby: '3.2'
+        ruby: ['3.4']
+        graphql: ['graphql-2.0.19', 'graphql-2.3']
```

Comment:
@RomKadria maybe worth adding `2.1` and `2.2`?

Reply:
I interntionally havnt as we dont use them